app/Http/Controllers/Api/TaskActivityController.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace App\Http\Controllers\Api;
+
+use App\Http\Controllers\Controller;
+use App\Http\Requests\TaskActivity\ShowTaskActivityRequest;
+use App\Models\TaskComment;
+use App\Models\TaskHistory;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Http\Request;
+use Illuminate\Pagination\LengthAwarePaginator;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\Auth;
+
+class TaskActivityController extends Controller
+{
+    protected $taskComment;
+    protected $taskHistory;
+
+    public function __construct(TaskComment $taskComment, TaskHistory $taskHistory)
+    {
+        $this->taskComment = $taskComment;
+        $this->taskHistory = $taskHistory;
+    }
+
+    public function getCollection(string $type,int $taskId): array
```

Comment:
For which purpose was this function made public? Do you have plans to use it outside of that controller?